src/api.v2/controllers/subsetController.js

```diff
@@ -10,66 +9,86 @@ const UserAccess = require('../model/UserAccess');
 
 const logger = getLogger('[SubsetController] - ');
 
-const runSubset = async (req, res) => {
-  const { experimentId } = req.params;
-
-  logger.log(`Starting subset for experiment ${experimentId}`);
-
+const runSubset = async (stateMachineParams, authorization) => {
   const {
-    params: { experimentId: parentExperimentId },
-    body: { name, cellSetKeys },
-    user: { sub: userId },
-  } = req;
+    experimentId: parentExperimentId,
+    name,
+    userId,
+  } = stateMachineParams;
 
-  logger.log(`Creating experiment to subset ${parentExperimentId}`);
+  // Doesn't exist initially
+  let { subsetExperimentId } = stateMachineParams;
 
-  let subsetExperimentId;
-  await sqlClient.get().transaction(async (trx) => {
-    subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
-    await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
+  // Upon retrying the subset experiment id already exists,
+  // and we don't want to create a new experiment
+  if (!subsetExperimentId) {
+    logger.log(`Creating experiment to subset ${parentExperimentId}`);
 
-    await new ExperimentParent(trx).create(
-      { experiment_id: subsetExperimentId, parent_experiment_id: parentExperimentId },
-    );
-  });
+    await sqlClient.get().transaction(async (trx) => {
+      subsetExperimentId = await new Experiment(trx).createCopy(parentExperimentId, name);
+      await new UserAccess(trx).createNewExperimentPermissions(userId, subsetExperimentId);
 
-  // Samples are not created here, we add them in handleResponse of SubsetSeurat
+      await new ExperimentParent(trx).create(
+        { experiment_id: subsetExperimentId, parent_experiment_id: parentExperimentId },
+      );
+    });
 
-  logger.log(`Created ${subsetExperimentId}, subsetting experiment ${parentExperimentId} to it`);
+    // Samples are not created here, we add them in handleResponse of SubsetSeurat
+    logger.log(`Created ${subsetExperimentId}, subsetting experiment ${parentExperimentId} to it`);
+  }
 
   const {
     processingConfig: parentProcessingConfig,
   } = await new Experiment().findById(parentExperimentId).first();
 
   const { stateMachineArn, executionArn } = await createSubsetPipeline(
-    parentExperimentId,
+    {
+      ...stateMachineParams,
+      subsetExperimentId,
+    },
     subsetExperimentId,
-    name,
-    cellSetKeys,
```

Comment:
Why was this param removed?

Reply:
They are still present, but because I need to persist them they are located in the `params` object.

Now I can see the issue with tractability, but here it is a trade-off between quality of code and readibility. Not sure which way to go, but I really like to keep the codebase tidy. Let me know what you think

Reply:
What about extracting them from `params` as soon as the functions runSubset, runSeurat, runGem2s begin?

Reply:
I see your point, but I would have to build the object again, before it gets persisted in the database. I am not sure if it is a good idea to unpack and pack it mutliple times.

Reply:
Hmm wouldn't you need to build it anyways? Since when you receive it, it is in camelcase and to persist in sql you need it in underscore

Reply:
No, I directly persist it to the db as an object. I believe this is okay and snake case only if things were columns in the table. This is an example entry in the `retry_params` column: 

```
{"experimentId": "547e6141-eca0-49b4-9b7d-7329a7649ddf"}
```

Reply:
Ah right, if they aren't columns then it's fine